lambda/main.tf

```
@@ -47,6 +47,15 @@ resource "aws_lambda_function" "this" {
     size = var.ephemeral_storage
   }
 
+  dynamic "file_system_config" {
+    for_each = length(var.file_system_config) == 0 ? [] : [true]
```

Can you switch this like so:
```suggestion
    for_each = length(keys(var.file_system_config)) == 0 ? [] : [var.file_system_config]
```
What this does is check if the `file_system_config` has any attributes, and if yes, creates a single item list of that `file_system_config` map object that gets looped over once by the `for_each` loop.